File: src/main/java/me/rsman/BetterMinecraftCore/BetterMinecraftCore.java
package me.rsman.BetterMinecraftCore;

import me.rsman.BetterMinecraftCore.Managers.*;
import me.rsman.BetterMinecraftCore.Managers.Command.CommandCompletionsManager;
import me.rsman.BetterMinecraftCore.Managers.Command.CommandManager;
import me.rsman.BetterMinecraftCore.configs.ConfigLoader;
import me.rsman.BetterMinecraftCore.configs.containers.BmcCraftContainer;
import org.bukkit.Bukkit;
import org.bukkit.plugin.java.JavaPlugin;

import java.util.logging.Level;
import java.util.logging.Logger;

public final class BetterMinecraftCore extends JavaPlugin {
    private static BetterMinecraftCore instance;

    @Override
    public void onEnable() {
        Logger pluginLogging = Logger.getLogger(this.getName());
        instance = this;

        getLogger().info("§aStarting");

        ConfigLoader.init();

        BmcCraftContainer.registerCrafts();

        DBManager.initConnection();

        TasksManager.registerAllTasks();
        ListenersManager.registerAllEvents();
        EnchantManager.registerAllEnchantments();
        //CraftManager.initCrafts();

        pluginLogging.setLevel(Level.SEVERE);
        CommandManager.init();
        CommandCompletionsManager.init();
        pluginLogging.setLevel(Level.INFO);


        getServer().getOnlinePlayers().forEach((player) -> {
            PlayerManager.getBaseAttributes(player.getUniqueId().toString(), true);
            PlayerManager.alterPlayerAttributesWithEquippedStuff(player);
        });

        getLogger().info("§aStarted");
    }

    @Override
    public void onDisable() {
        // Plugin shutdown logic
        getLogger().info("§aShutting down");
        EnchantManager.unRegisterAllEnchantments();
        getServer().resetRecipes();
    }


    public static BetterMinecraftCore getInstance() {
        return instance;
    }
}

File: src/main/kotlin/me/rsman/BetterMinecraftCore/BetterMinecraftCore.kt
package me.rsman.BetterMinecraftCore

import me.rsman.BetterMinecraftCore.Managers.*
import org.bukkit.plugin.java.JavaPlugin
import me.rsman.BetterMinecraftCore.configs.ConfigLoader
import me.rsman.BetterMinecraftCore.configs.containers.BmcCraftContainer
import me.rsman.BetterMinecraftCore.Managers.Command.CommandCompletionsManager
import me.rsman.BetterMinecraftCore.Managers.Command.CommandManager
import org.bukkit.entity.Player
import java.util.logging.Level
import java.util.logging.Logger

class BetterMinecraftCore : JavaPlugin() {
    override fun onEnable() {
        val pluginLogging = Logger.getLogger(this.name)
        instance = this
        logger.info("§aStarting")
        ConfigLoader.init()
        BmcCraftContainer.registerCrafts()
        DBManager.initConnection()
        TasksManager.registerAllTasks()
        ListenersManager.registerAllEvents()
        EnchantManager.registerAllEnchantments()
        //CraftManager.initCrafts();
        pluginLogging.level = Level.SEVERE
        CommandManager.init()
        CommandCompletionsManager.init()
        pluginLogging.level = Level.INFO
        server.onlinePlayers.forEach { player: Player? ->
            PlayerManager.getBaseAttributes(player!!.uniqueId.toString(), true)
            PlayerManager.alterPlayerAttributesWithEquippedStuff(player)
        }
        PapiManager.registerExpansion()
        logger.info("§aStarted")
    }

    override fun onDisable() {
        // Plugin shutdown logic
        logger.info("§aShutting down")
        EnchantManager.unRegisterAllEnchantments()
        server.resetRecipes()
    }

    companion object {
        @JvmStatic
        lateinit var instance: BetterMinecraftCore
            private set
    }
}